### PR TITLE
da1469x_pd: add function to get power domain reference counts

### PR DESCRIPTION
**Pull request by JustineKH**
da1469x_pd.c: Add function to get power domain reference counts.
hal_gpio.c: Restore Px_DATA_REG from retained values. Seeing issues/hangs when attempting to restore it from itself, when it is possibly not retained when PD_COM is down.

### CODE REVIEW

**Review by wes3 — APPROVED** [review]
I approve of the changes but my preference would be to return the native type of an element in a structure if you are returning that element. Not sure everyone would agree with that btw. Returning and int will work but my preference would be that this return uint8_t. You could also have done the following and saved some typing but no need to change :-)

assert(pd < PD_COUNT);
return g_da1469x_pd_data[pd].refcnt;

Anyway, all good and no need to change anything.

**Comment by andrzej-kaczmarek** [comment]
adding new API is fine, although I only see it being used to make workarounds like in the 2nd patch which should not be merged upstream unless this fix can be explained.

File: hw/mcu/dialog/da1469x/src/hal_gpio.c [hunk 1]
```diff
@@ -473,6 +474,10 @@ mcu_gpio_enter_sleep(void)
         return;
     }
 
+    /* Save pins states and their latched values */
```
Comment (benmccrea) [inline hw/mcu/dialog/da1469x/src/hal_gpio.c:477]: @JustineKH Are interrupts already disabled when `mcu_gpio_enter_sleep` gets called? If not,  an interrupt could change pin levels before entering sleep.

Comment (andrzej-kaczmarek) [reply hw/mcu/dialog/da1469x/src/hal_gpio.c:477]: the whole sleep code is run with interrupts disabled since it's called from 'os_tick_idle`

**Pull request merged**